### PR TITLE
Update HttpCache kernel documentation for Symfony 4

### PR DESCRIPTION
**Pull request by michaelperrin**
#SymfonyConHackday2017
I am not entirely sure that we should document this way (creating a `src/CacheKernel.php` file as I propose here). Feel free to comment :)

### CODE REVIEW

**Comment by javiereguiluz** [comment]
I guess this is duplicated of #8682 ... these days we're updating all docs, so please check the open PR and create an issue saying that you are going to work on some files. Thanks!

**Comment by michaelperrin** [comment]
Thanks @javiereguiluz . That's too bad because I checked but didn't see your PR, my bad.

The cool thing is that @weaverryan proposes the same name for the caching kernel, so it seems its quite a natural one :)

You can close this PR then if you wish!

File: http_cache.rst [hunk 1]
```diff
@@ -77,23 +77,29 @@ but is a great way to start.
...
 
-Enabling the proxy is easy: each application comes with a caching kernel (``AppCache``)
-that wraps the default one (``AppKernel``). The caching Kernel *is* the reverse
-proxy.
+To enable the proxy, first create a caching kernel::
 
-To enable caching, modify the code of your front controller. You can also make these
-changes to ``index.php`` to add caching to the ``dev`` environment::
+    // src/CacheKernel.php
+    <?php
```
Comment (weaverryan) [inline http_cache.rst]: remove the `<?php`

File: http_cache.rst [hunk 2]
```diff
@@ -77,23 +77,29 @@ but is a great way to start.
...
+    use Symfony\Bundle\FrameworkBundle\HttpCache\HttpCache;
+
+    class CacheKernel extends HttpCache
+    {
+    }
+
+Modify the code of your front controller to wrap the default kernel into the
+caching kernel::
 
     // public/index.php
```
Comment (weaverryan) [inline http_cache.rst]: Let's change the `::` to just `:` and add (on the next line) `.. code-block:: diff`.

Then, you can add + and - to emphasize the changes:

```
.. code-block:: diff

    // some line from before
    - // line to go away
    + // line to add
```

**Review by weaverryan — CHANGES_REQUESTED** [review]
@michaelperrin if you are able to make the suggested changes faster than Javier, then I would be happy to merge your's in. Javier's PR has additional changes - so he could just rebase after your merge. But, quick! Symfony 4 is upon us! :)

File: http_cache.rst (hunk 2 above)
Comment (michaelperrin) [reply http_cache.rst]: That's great advice, thanks :) I could check that everything is fine by compiling the doc with Sphinx.

**Comment by michaelperrin** [comment]
@weaverryan I made the changes you requested :)

**Comment by weaverryan** [comment]
Thank you Michaël!

**Pull request merged**